streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordDeserializer.java

```diff
@@ -78,6 +79,10 @@
                 throw new StreamsException("Fatal user code error in deserialization error callback", fatalUserException);
             }
 
+            if (deserializationException instanceof ConfigException) {
```

Comment:
Should this check be done before we call `deserializationExceptionHandler.handle` ?

Reply:
I think it could be, but I'm not sure I understand why that might be preferable. I guess we wouldn't want the config exception overridden by a user exception? In that case,  yeah it would make sense to move it up. When I saw this in testing the first check for user error didn't interfere with seeing the config error, but I can see how that would change when running a full app

Reply:
My thought was, that for a `ConfigException` we are doomed to fail anyway, and thus it seems not make sense to call the handler to allow the user to "swallow" the exception by returning `CONTINUE` ?

Also, even if the user does return `CONTINUE`, it seems we could ignore it and rethrow the `ConfigException` and die anyway, what seems to defeat the purpose of (calling) the exception handler to begin with?

Reply:
Makes sense, I'll shift around